curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md

```diff
@@ -11,7 +11,21 @@ So far we have learned to use pure functions to avoid side effects in a program.
 This is only the beginning. As its name suggests, functional programming is centered around a theory of functions.
 It would make sense to be able to pass them as arguments to other functions, and return a function from another function. Functions are considered <dfn>first class objects</dfn> in JavaScript, which means they can be used like any other object. They can be saved in variables, stored in an object, or passed as function arguments.
 Let's start with some simple array functions, which are methods on the array object prototype. In this exercise we are looking at <code>Array.prototype.map()</code>, or more simply <code>map</code>.
-Remember that the <code>map</code> method is a way to iterate over each item in an array. It creates a new array (without changing the original one) after applying a callback function to every element.
+The <code>map</code> method iterates over each item in an array and returns a new array containing the results of calling the callback function on each element.  It does this without mutating the original.
+When the callback is used, it is passed three arguments.  The first argument is the current element being processed.  The second is the index of that element and the third is the array upon which the <code>map</code> method was called.
+See below for an example using the <code>map</code> method on the <code>users</code> array to return a new array containing only the names of the users as elements. For simplicity, the example only uses the first argument of the callback.
+
+```js
+const users = [
+  { name: 'John', age: 34 },
+  { name: 'Amy', age: 20 },
+  { name: 'camperCat', age: 10 }
+];
+
+const names = users.map(user => user.name);
+console.log(names); // [ 'John', 'Amy', 'camperCat' ]
+```
+
 </section>
 
 ## Instructions
```

Comment:
```
Use <code>map</code> to pull the title and rating from <code>watchList</code> and save the new array in the <code>rating</code> variable.
```

This instruction should probably be changed. It sounds like we want users to pull the title and rating properties directly into a new array. While the task is actually to return a new "object" from the `.map()`, with `title` and `rating` properties, containing the corresponding values from each object. This array of new objects should be saved in the `watchList` variable.

Yes, there is the for-loop in the example, which you could probably look at to deduce this, but instructions should ideally be self-explanatory.

Reply:
@Manish-Giri I made a new commit that I think improves the instructions for this challenge.  Please let me know what you think.